src/rdd.rs

```diff
@@ -153,6 +174,38 @@ pub trait Rdd<T: Data>: RddBase + Send + Sync + Serialize + Deserialize {
         self.get_context().run_job_with_context(self.get_rdd(), cl);
     }
 
+    fn reduce<F>(&self, f: F) -> Option<T>
+    where
+        Self: Sized + 'static + Serialize + Deserialize,
+        F: Fn(T, T) -> T
+            + PartialEq
+            + Eq
+            + Clone
+            + Send
+            + Sync
+            + serde::ser::Serialize
+            + serde::de::DeserializeOwned
+            + 'static,
+    {
+        // cloned cause we will use `f` later.
+        let cf = f.clone();
+        let reduce_partition = Fn!([cf] move |iter: Box<dyn Iterator<Item = T>>| {
+        let acc = iter.reduce(cf);
+        match acc {
+            None => vec![],
+            Some(e) => vec![e],
+        }
+
+        });
+        let results = self.get_context().run_job(self.get_rdd(), reduce_partition);
+        results
+            .into_iter()
+            .flatten()
+            .collect::<Vec<T>>()
```

Comment:
This intermediate allocation is not necessary I think. flatten().reduce(f) itself should do the same work. If there is something I overlooked, let me know. Otherwise it looks fine.

Reply:
fixed